src/nbl/video/IGPUCommandBuffer.cpp

```diff
@@ -1774,6 +1798,36 @@ bool IGPUCommandBuffer::resolveImage(const IGPUImage* const srcImage, const IGPU
     return resolveImage_impl(srcImage, srcImageLayout, dstImage, dstImageLayout, regionCount, pRegions);
 }
 
+bool IGPUCommandBuffer::traceRays(const asset::SStridedBufferRegion<IGPUBuffer>& raygenGroupRegion,
+    const asset::SStridedBufferRegion<IGPUBuffer>& missGroupsRegion,
+    const asset::SStridedBufferRegion<IGPUBuffer>& hitGroupsRegion,
+    const asset::SStridedBufferRegion<IGPUBuffer>& callableGroupsRegion,
+    uint32_t width, uint32_t height, uint32_t depth)
+{
+    if (!checkStateBeforeRecording(queue_flags_t::COMPUTE_BIT,RENDERPASS_SCOPE::OUTSIDE))
+        return false;
+
+    if (width == 0 || height == 0 || depth == 0)
+    {
+        NBL_LOG_ERROR("invalid work counts (%d, %d, %d)!", width, height, depth);
+        return false;
+    }
+
+    if (!m_cmdpool->m_commandListPool.emplace<IGPUCommandPool::CTraceRaysCmd>(m_commandList, 
```

Comment:
add checks for:
https://registry.khronos.org/vulkan/specs/latest/html/vkspec.html#VUID-vkCmdTraceRaysKHR-pRayGenShaderBindingTable-03681
https://registry.khronos.org/vulkan/specs/latest/html/vkspec.html#VUID-vkCmdTraceRaysKHR-pRayGenShaderBindingTable-03682
https://registry.khronos.org/vulkan/specs/latest/html/vkspec.html#VUID-vkCmdTraceRaysKHR-stride-03690

and friends

TL;DR check that:
- buffers have SBT usages
- their base address (BDA + offset) are aligned to `shaderGroupBaseAlignment`
- the stride (if any) is aligned to `shaderGroupHandleAlignment`
- the stride is less than `maxShaderGroupStride`

and leave comments with hyperlinks to VUIDs from the spec (see the beautiful validation of renderpasses).